lightning/src/chain/keysinterface.rs

```diff
@@ -245,12 +248,92 @@ pub trait ChannelKeys : Send+Clone {
+	/// detected onchain. It has been generated by remote party and is used to derive
+	/// channel state keys, committed as part of witnessScript by sigs (BIP 143).
+	fn sign_remote_htlc_transaction<T: secp256k1::Signing + secp256k1::Verification>(&self, htlc_tx: &Transaction, input: usize, amount: u64, per_commitment_point: &PublicKey, htlc: &HTLCOutputInCommitment, secp_ctx: &Secp256k1<T>) -> Result<Signature, ()>;
+
 	/// Create a signature for a (proposed) closing transaction.
 	///
 	/// Note that, due to rounding, there may be one "missing" satoshi, and either party may have
 	/// chosen to forgo their output as dust.
 	fn sign_closing_transaction<T: secp256k1::Signing>(&self, closing_tx: &Transaction, secp_ctx: &Secp256k1<T>) -> Result<Signature, ()>;
 
+	/// Create a signature for a delayed transaction, spending a local commitment/HTLC output.
+	///
+	/// Output spending is constrained bya OP_CSV and therefore spending must set accurately
+	/// its nSequence to satisfy encumbrance.
+	///
+	/// It may be called multiples time for same output(s) if a fee-bump is needed but there
+	/// is no unfavorable timelock expiration to consider. Therefore it doesn't have to
+	/// be implemented on the same Lightning channel signer and can be on a "cooler" host.
+	///
+	/// Input index to point towards outpoint spent, committed in sig (BIP143).
+	///
+	/// CSV is the relative lock-time value as expected to be part of witnessScript.
+	///
+	/// Amount is value of the output spent by this input, committed in sig (BIP143).
+	///
+	/// Per_commitment_point is the dynamic point corresponding to the channel state
+	/// detected onchain. It has been generated by us and is used to derive our
+	/// delayed_payment_key.
+	fn sign_delayed_transaction<T: secp256k1::Signing + secp256k1::Verification>(&self, spend_tx: &Transaction, input: usize, on_local_tx_csv: u16, amount: u64, per_commitment_point: &PublicKey, secp_ctx: &Secp256k1<T>) -> Result<Signature, ()>;
+
+	/// Create a signature for a payment transaction spending a to_remote output on a remote commitment tx.
+	///
+	/// Output spending is a simple P2WPKH and doesn't have any spending constrain beyond
+	/// signature verification.
+	///
+	/// It may be called multiples time for same output(s) if a fee-bump is needed but there
+	/// is no unfavorable timelock expiration to consider. Therefore it doesn't have to
+	/// be implemented on the same Lightning channel signer and can be on a "cooler" host.
+	///
+	/// Input index to point towards outpoint spent, committed in sig (BIP143).
+	///
+	/// Amount is value of the output spent by this input, committed in sig (BIP143).
+	///
+	/// Per_commitment_point is the dynamic point corresponding to the channel state
+	/// detected onchain.
+	fn sign_payment_transaction<T: secp256k1::Signing>(&self, spend_tx: &Transaction, input: usize, amount: u64, per_commitment_point: &PublicKey, network: Network, secp_ctx: &Secp256k1<T>) -> Result<Signature, ()>;
```

Comment:
I'm confused - we never call this outside of tests, why is it in a public interface?

Reply:
Removed